### PR TITLE
DEVPROD-6820 Change git clone error log from info level to error level

### PR DESCRIPTION
**Pull request by ZackarySantana**
DEVPROD-6820

### Description
Changes git clone error log from info level to error level

### CODE REVIEW

File: agent/command/git.go [hunk 1]
```diff
@@ -787,7 +787,7 @@ func (c *gitFetchProject) fetchModuleSource(ctx context.Context,
 			if opts.token != "" {
 				errOutput = strings.Replace(errOutput, opts.token, "[redacted oauth token]", -1)
 			}
-			logger.Execution().Info(errOutput)
+			logger.Execution().Error(errOutput)
```
Comment (malikchaya2) [inline agent/command/git.go:790]: I like your idea in the ticket of adding the module name. Should we maybe wrap the error and include module.Name?

Comment (ablack12) [reply agent/command/git.go:790]: I'm maybe confused about how this prevents the output from being jumbled

**Review by ablack12 — CHANGES_REQUESTED** [review]

File: agent/command/git.go (hunk 1 above)
Comment (ZackarySantana) [reply agent/command/git.go]: It doesn't make it not jumbled, but it made it easier to see with a red line in parsley...

I changed it now to collect in a buffer and display it one after another with every new line having the module name, and the error having the module name fast.

**Review by ablack12 — APPROVED** [review]
LGTM, you'll need to increment the agent version string (so probably also merge main)

**Review by malikchaya2 — APPROVED** [review]

**Pull request merged**